src/liballoc/vec.rs

```diff
@@ -1,3 +1,4 @@
+// ignore-tidy-filelength
```

Comment:
I'm not quite sure what's the procedure on this and what this tidy check tries to achieve, but it feels like this file could easily be split. There are already multiple sections. It would be easy, for example, to move the iterator stuff out of there (the last section, starting [here](https://github.com/rust-lang/rust/blob/3477e67a92878adae48b975915cb7a5c21026cd4/src/liballoc/vec.rs#L2486-L2488)). It's kind of unfortunate that you have to do it now, since this is unrelated to your PR, but I'd prefer that over the ignore tidy annotation.

But just to get a second opinion: @BurntSushi @withoutboats What do you think?

Reply:
The purpose of the tidy check is to avoid gigantic files with poor structure. I think doing it before this PR or after is fine, but someone should sign up for the work.